### PR TITLE
[CoW] Migrate Trades Table

### PR DESCRIPTION
**Pull request by bh2smith**
Migration of CoW Protocol Trades table. This was a bit of a big one, but somehow trades is a relatively low level table with very few dependencies. This should pave the way to migrate nearly all the rest of the cow spells.

cc @gentrexha

### CODE REVIEW

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 1]
```diff
@@ -130,48 +125,52 @@ orders_and_trades as (
...
-        lateral view posexplode(order_ids) o as i, uid
-        lateral view posexplode(trades) t as j, trade
-    where i = j
+    SELECT
+      uid,
+      from_hex(JSON_EXTRACT_SCALAR(trade, '$.appData')) AS app_data,
+      from_hex(JSON_EXTRACT_SCALAR(trade, '$.receiver')) AS receiver,
+      cast(JSON_EXTRACT_SCALAR(trade, '$.sellAmount') as uint256) AS limit_sell_amount,
+      cast(JSON_EXTRACT_SCALAR(trade, '$.buyAmount') as uint256) AS limit_buy_amount,
+      JSON_EXTRACT_SCALAR(trade, '$.validTo') AS valid_to,
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:137]: What is the content of this column? Could it have another data type than varchar?

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 2]
```diff
@@ -130,48 +125,52 @@ orders_and_trades as (
...
 
 eth_flow_senders as (
     select
         sender,
-        concat(output_orderHash, substring(event.contract_address, 3, 40), 'ffffffff') as order_uid
+        from_hex(concat(
+            cast(output_orderHash AS VARCHAR),
+            substring(cast(event.contract_address as varchar), 3, 40),
+            'ffffffff'
+        )) AS order_uid
```
Comment (jkylling) [inline models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:158]: These casts can be avoided by the use of bytearray_concat and bytearray_substring

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql (hunk 1 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:137]: Yea probably. I think it's a Unix time stamp. Will look into it and get it right.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql (hunk 2 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:158]: Thanks a bunch. Will do.

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql (hunk 1 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql]: Fixed!

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql (hunk 2 above)
Comment (bh2smith) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql]: Done.

**Review by jeff-dude — APPROVED** [review]
great stuff ser -- thank you 🙌

**Review by gentrexha — COMMENTED** [review]
I actually had this reviewed but never submitted. Would've caught the block_number missing as well. Nonetheless, thanks for migrating this Ben!

File: models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql [hunk 3]
```diff
@@ -39,28 +39,27 @@ trades_with_prices AS (
...
                                  {% endif %}
     {% if is_incremental() %}
-    WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+    WHERE evt_block_time >= date_trunc("day", now() - interval '7' day)
     {% endif %}
 ),
 -- Second subquery gets token symbol and decimals from tokens.erc20 (to display units bought and sold)
 trades_with_token_units as (
     SELECT block_date,
-           block_number,
```
[the remark this answers is not in the document]
Comment (gentrexha) [reply models/cow_protocol/ethereum/cow_protocol_ethereum_trades.sql:63]: This has been already caught and resolved by Ben.

**Pull request merged**